### PR TITLE
Add a link to the user profile in chats

### LINKED ISSUE #4501: Add ability to click someones name in chat to go to their profile (web chat)
I'd like the "Talking with username" to be clickable to go to their profile, either in a new tab or in the same tab.
random example of what i'm talking about:
[image](https://i.gyazo.com/6299e14155bfb94d19c99b7629a2f90c.png)
link to the chat:
https://osu.ppy.sh/community/chat

### PR DESCRIPTION
**Pull request by LiquidPL**
Fixes #4501.

~~Also since this changes an existing translation, what's the process with the non-english ones? Should I change them in the PR as possible, or just leave them out for crowdin translators to update?~~ not anymore

### CODE REVIEW

File: resources/assets/lib/chat/conversation-view.tsx [hunk 1]
```diff
@@ -147,7 +147,7 @@ export default class ConversationView extends React.Component<any, any> {
         </div>
         <div className='chat-conversation__chat-label'>
           {channel.type === 'PM' ? (
-            osu.trans('chat.talking_with', {name: channel.name})
+            <span>{osu.trans('chat.talking_with')} <a href={laroute.route('users.show', {user: channel.pmTarget})}>{channel.name}</a></span>
```
Comment (notbakaneko) [inline resources/assets/lib/chat/conversation-view.tsx]: not all languages have the same order

File: resources/assets/lib/chat/conversation-view.tsx [hunk 2]
```diff
@@ -147,7 +147,9 @@ export default class ConversationView extends React.Component<any, any> {
         </div>
         <div className='chat-conversation__chat-label'>
           {channel.type === 'PM' ? (
-            <span>{osu.trans('chat.talking_with')} <a href={laroute.route('users.show', {user: channel.pmTarget})}>{channel.name}</a></span>
+            <span dangerouslySetInnerHTML={{__html: osu.trans('chat.talking_with', {
```
Comment (notbakaneko) [inline resources/assets/lib/chat/conversation-view.tsx]: use the new `StringWithComponent` instead

**Comment by LiquidPL** [comment]
I also wanted to use the `UserLink` component because usercards, but there's this mess with the user interfaces:

```
<UserLink key='user' user={{id: channel.pmTarget || 0, username: channel.name, cover: {}, default_group: '', is_active: false, is_bot: false, is_online: false, is_supporter: false, pm_friends_only: false}}/>
```

What should I do here?

**Comment by notbakaneko** [comment]
use `<a className='js-usercard' data-user-id=` directly instead of `UserLink` for now

**Pull request merged**